pkg/initializers/dataset/cache.py

```diff
@@ -0,0 +1,278 @@
+                                        "command": ["head"],
+                                        "args": ["0.0.0.0", "50051"],
+                                        "resources": {
+                                            "limits": {
+                                                "cpu": head_cpu,
+                                                "memory": head_mem,
+                                            },
+                                            "requests": {
+                                                "cpu": head_cpu,
+                                                "memory": head_mem,
+                                            },
+                                        },
+                                        "env": env_vars,
+                                        "ports": [{"containerPort": 50051}],
+                                    }
+                                ],
+                            },
+                        },
+                        "workerTemplate": {
+                            "spec": {
+                                "serviceAccountName": f"{train_job_name}-sa",
+                                "containers": [
+                                    {
+                                        "name": "worker",
+                                        "image": cache_image,
+                                        "command": ["worker"],
+                                        "args": ["0.0.0.0", "50051"],
+                                        "resources": {
+                                            "limits": {
+                                                "cpu": worker_cpu,
+                                                "memory": worker_mem,
+                                            },
+                                            "requests": {
+                                                "cpu": worker_cpu,
+                                                "memory": worker_mem,
+                                            },
+                                        },
+                                        "env": env_vars,
+                                        "ports": [{"containerPort": 50051}],
+                                    }
+                                ],
+                            }
+                        },
+                    },
+                },
+            }
+
+            # Create LeaderWorkerSet
+            custom_api.create_namespaced_custom_object(
+                group="leaderworkerset.x-k8s.io",
+                version="v1",
+                namespace=namespace,
+                plural="leaderworkersets",
+                body=lws_body,
+            )
+            logging.info(f"Created LeaderWorkerSet {lws_body['metadata']['name']}")
+
+            # Create Service
+            service = client.V1Service(
+                metadata=client.V1ObjectMeta(
+                    name=f"{train_job_name}-cache-service",
```

Comment:
```suggestion
                    name=f"{train_job_name}-cache",
```

Reply:
I will have to revert this change to cache-service for now. It is conflicting with the default headless service created by lws controller with same name as the lws app.